PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/test/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/HardcodedAPIKeysAndTokensCheckTest.java

```diff
@@ -0,0 +1,134 @@
+package com.microsoft.azure.toolkit.intellij.azure.sdk.buildtool;
+import com.microsoft.azure.toolkit.intellij.azure.sdk.buildtool.HardcodedAPIKeysAndTokensCheck.APIKeysAndTokensVisitor;
+
+import com.intellij.psi.JavaElementVisitor;
+import com.intellij.psi.PsiJavaCodeReferenceElement;
+import com.intellij.psi.PsiNewExpression;
+
+import com.intellij.codeInspection.ProblemsHolder;
+import com.intellij.psi.PsiElementVisitor;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * Test the HardcodedAPIKeysAndTokensCheck class for hardcoded API keys and tokens.
+ * When a client is authenticated with AzurekeyCredentials and AccessToken, a problem is registered.
+ These are some instances that a flag would be raised.
+ * 1. TextAnalyticsClient client = new TextAnalyticsClientBuilder()
+ *         .endpoint(endpoint)
+ *         .credential(new AzureKeyCredential(apiKey))
+ *         .buildClient();
+ *
+ * 2. TokenCredential credential = request -> {
+ *         AccessToken token = new AccessToken("<your-hardcoded-token>", OffsetDateTime.now().plusHours(1));
+ *     }
+ */
+public class HardcodedAPIKeysAndTokensCheckTest {
+
+    // Declare as instance variables
+    @Mock
+    private ProblemsHolder mockHolder;
+
+    @Mock
+    private PsiElementVisitor mockVisitor;
+
+    @BeforeEach
+    public void setup() {
+        mockHolder = mock(ProblemsHolder.class);
+        mockVisitor = createVisitor();
+    }
+
+
+    /**
+     * Test the HardcodedAPIKeysAndTokensCheck class for hardcoded API keys and tokens.
+     * When a client is authenticated with AzurekeyCredentials and AccessToken, a problem is registered.
+     */
+    @Test
+    public void testHardcodedAPIKeysAndTokensCheck() {
```

Comment:
Use ParameterizedTest to test all the variations of authentications we need to flag.
https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests

Reply:
done:)